engine/common/multiplexer/network.go

```diff
@@ -0,0 +1,53 @@
+package multiplexer
+
+import (
+	"fmt"
+
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/network"
+)
+
+type Network struct {
+	net      module.Network
+	eng      Engine
+	conduits map[network.Channel]network.Conduit // stores conduits for all registered channels
+}
+
+func NewNetwork(
+	net module.Network,
+	eng Engine,
+) (*Network, error) {
+	e := &Network{
+		net:      net,
+		eng:      eng,
+		conduits: make(map[network.Channel]network.Conduit),
+	}
+
+	return e, nil
+}
+
+// Register will subscribe the given engine with the multiplexer on the given channel, and all registered
+// engines will be notified with incoming messages on the channel.
+// The returned Conduit can be used to send messages to engines on other nodes subscribed to the same channel
+func (n *Network) Register(channel network.Channel, engine network.Engine) (network.Conduit, error) {
```

Comment:
⚠️ ❗ ⚠️ 
Would strongly suggest to make this concurrency safe.

Reply:
https://github.com/onflow/flow-go/pull/947/commits/5b5fe77b97b5d5fbca7fe049c351126f7d0b0604